### ISSUE TITLE
PDC files associated with parent specimen rather than the specific specimen analyzed

### ISSUE THREAD

**Issue by ianfore** [CancerDataAggregator/cda-python#97]
Am retrospectively adding issues that were reported in CDA v1 but not entered as a GitHub issue.
This was issue 4 from the 09CO022 Example notebook.
The ground truth model referred to below is shown in that notebook.

This particular issue may have been fixed in V2, but still recording it in recognition of that, and confirming it is the case.

**Describe the bug**
Specimens went to locations more than 2000 miles apart, so they must have been physically separate materials. The ground truth model shows that - with the specimens having visibly different ids. 
Per the ground truth model these specimens were known as
76498650-fdbf-4f5d-a19a-cce9a2_D2 (Normal specimen sent to PNNL)
f69deaeb-6b6f-4c61-8900-fd0f26_D2 (Tumor specimen sent to VU)
f69deaeb-6b6f-4c61-8900-fd0f26_D3 (Tumor specimen sent to PNNL)
That has been lost in the PDC/CDA representation. The specimens in CDA represents the parent not the physically distinct samples at each site.

**To Reproduce**
See the notebook referred to above

**Expected behavior**
The files from each specimen should only be associated with the correct specimen from the list above.

**Comment by druth314** [comment]
In R1 we only recorded files associated with samples(all files associated with children of that sample). In R2 we now have files associated with all specimen types (samples, portions, slides, analytes, and aliquots). Each parent specimen has records of files associated with itself, and all of it's children. For example if it is from GDC, any files associated with an aliquot, will also be recorded for the parent analyte, portion, and sample from which it is derived.

**Comment by ianfore** [comment]
Hoping this reopens the issue. 

First of all, v2 does fix the issue by associating the file with the specific aliquot. As covered in this [notebook](https://github.com/ianfore/cdatest/blob/main/v2tests/09CO022%20Example%20-%20CDA%20v2.ipynb).

However, the repetition of the files at the higher level confuses things. The solution where the file is only associated with its specific specimen makes it explicit what that file means i.e. its semantics. The file is still nested within the higher level - so it is still visible as part of the sample as well as the aliquot.

**Comment by druth314** [comment]
This is true if we had explicitly kept the hierarchical structure within the Specimen entity(samples: [ {portions: [{ analytes...) but we have essentially flattened that hierarchical structure and just have linking columns pointing to parent Specimens. This was a conscious choice by CCDH that we are implementing. As far as associating files of child specimens with the parents, we figured it would be easier for users to have this repeated structure so they would not have to try to aggregate all files from the child specimens. This can be re-examined and discussed though.

**Comment by ianfore** [comment]
Thanks Donovan, particularly the openness to revisiting and discussing.

I could have been clearer about nesting. The specimens are certainly not nested within one another in json but the real world nesting/hierarchy is still there via the link to the parent in derived_from_specimen. The important change in the CCDH model was the flexible depth instead of the fixed hierarchy that worked for TCGA but doesn't have general applicability. 

The simplification that it's easier for users to have the repeated structure at the higher level is deceptive. To make any meaningful use of the files you have to know their particular place in the study. Did it come from the tumor or the normal? Did the histopathology come from the same portion as the genomic data? etc. That last one is scientifically important because it says something about whether it's reasonable to correlate the pathology with the genomics.

The true place of the file in the hierarchy is therefore important, and as a user one is better served by knowing that.

There are other aspects of this that I began to explore in v1 - particularly with the proteomics data. The [09CO022 Proteomic Files](https://github.com/ianfore/cdatest/blob/main/v1tests/09CO022%20Proteomic%20Files.ipynb) notebook worked through it, but it could do with better commentary to draw out the significance. I'll rework that through for v2 and add issues here where relevant. The same underlying issues are showing up with the IDC files. I'm finishing up a notebook on that and will post findings shortly. It's worth working through these specific cases separately, but there are some core issues which will have power because of their general applicability.

This will get more complicated as highly advanced studies like HTAN come on board with a more complicated structure. So getting the core right in anticipation is important.

**Comment by druth314** [comment]
I completely understand that particular use case, and having the ability to specifically have files associated with the most direct specimens. This will need to be a longer conversation with our stakeholders as well as CCDH because both GDC and PDC reflect the kind of association with their API's that we present. Part of our responsibility is to help join data together from different DC's and implement a data structure that meshes well with CCDH's data model, and another right now is to present data as we find it from the DC's since we are not fully responsible for the harmonization of this data. I can add this note to discuss with CCDH and see what they say. We are working on some changes in regards to features and potential architecture changes that will heavily affect how File information is accessed and stored.
